tests/foreman/ui/test_contentview.py

```diff
@@ -3967,3 +3967,291 @@ def test_negative_non_readonly_user_actions(module_org, test_name):
+
+
+@tier2
+def test_positive_greedy_solve_dependencies(session, module_org):
+    """Performing solve dependencies on a package that is required by another
+    package.  Then performing solve dependencies on a root package with
+    its corresponding dependency.
+
+    :id: 2b588667-d262-409b-8500-a4fb9092ce2b
+
+    :setup:
+        1. Turn dependency solving to greedy.
+        2. Create and sync custom repo.
+        3. Create cv and add custom repo.
+        4. Create exclusion filter and exclude a package that is required
+        by another package (cockateel).
+        5. Publish CV
+        6. Check rpm package list for that version and make sure it's in the
+        list.
+        7. Create exclusion filter for a root package (duck) along with
+        package from above.
+        8. Republish and check rpm package list for new version and make
+        sure they are NOT in the list.
+
+    :expectedresults: In step 6, 'cockateel' should be in the list.
+        In step 8, 'duck' and 'cockateel' should NOT be in the list.
+
+    :CaseImportance: High
+    """
+    property_name = 'dependency_solving_algorithm'
+    param_value = 'greedy'
+    cv_name = gen_string('alpha')
+    repo_name = gen_string('alpha')
+    package1_name = 'duck'
+    package2_name = 'cockateel'
+    arch = 'noarch'
+    create_sync_custom_repo(
+        module_org.id,
+        repo_name=repo_name,
+        repo_url=FAKE_0_YUM_REPO
+    )
+    with session:
+        session.settings.update(
+                'name = {}'.format(property_name),
+                param_value
+        )
+        session.contentview.create({
+            'name': cv_name,
+            'solve_dependencies': True,
+        })
+        session.contentview.add_yum_repo(cv_name, repo_name)
+        filter_name = gen_string('alpha')
+        session.contentviewfilter.create(cv_name, {
+            'name': filter_name,
+            'content_type': FILTER_CONTENT_TYPE['package'],
+            'inclusion_type': FILTER_TYPE['exclude'],
+        })
+        session.contentviewfilter.add_package_rule(
+            cv_name, filter_name, package2_name, arch, ('All Versions'))
+        result = session.contentview.publish(cv_name)
+        assert result['Version'] == VERSION
+        packages = session.contentview.search_version_package(
+            cv_name, VERSION, 'name = "{}"'.format(package2_name))
+        assert len(packages) == 1
+        assert packages[0]['Name'] == package2_name
+        session.contentviewfilter.add_package_rule(
+            cv_name, filter_name, package1_name, arch, ('All Versions'))
+        result = session.contentview.publish(cv_name)
+        assert result['Version'] == 'Version 2.0'
+        for package_names in ['duck', 'cockateel']:
+            packages = session.contentview.search_version_package(
+                cv_name, 'Version 2.0', 'name = "{}"'.format(package_names))
+            assert not packages[0]['Name']
+
```

Comment:
@latran Just to understand more both algorithm solves the problem with the same output. Is there any difference in performance?  Or any difference we can catch on UI or System Level?

Reply:
There's no differences in performances.   I could've stacked them into one test case and change the settings to greedy/conservatives but I thought splitting it up would be a better way to distinguish them.  That's why you see 2 of the same results.  Any advice on whether I should delete this test case or redo it or leave it as is?

Reply:
Ok I just wanted to understand the feature. If you think you can merge or put it common code and runs over algorithms that is fine too. 

I'm good even you want to keep it separate. Not a blocker issue for me.